linera-rpc/src/node_provider.rs

```diff
@@ -10,11 +15,12 @@ use crate::{client::Client, grpc::GrpcNodeProvider};
 
 /// A general node provider which delegates node provision to the underlying
 /// node provider according to the `ValidatorPublicNetworkConfig`.
-#[derive(Copy, Clone)]
+#[derive(Clone)]
 pub struct NodeProvider {
     grpc: GrpcNodeProvider,
     #[cfg(with_simple_network)]
     simple: SimpleNodeProvider,
+    client_pool: Arc<Mutex<HashMap<String, Client>>>,
```

Comment:
Based on the experience of #2664, the `GrpcNodeProvider` should instead include a `HashMap<String, tonic::transport::Channel>`